src/jsonata/parser.py

```diff
@@ -1245,6 +1246,7 @@ def lambda4(arg):
 
             result.arguments = [lambda4(x) for x in expr.arguments]
             result.procedure = self.process_ast(expr.procedure)
+            result.keep_array = result.procedure.keep_array
```

Comment:
This line seems unnecessary, the tests still pass without it.  I believe that's because  of 
```
        if expr.keep_array:
            result.keep_array = True
```
at the end of the function

Reply:
removed that change.